hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/rmadmin/RMAdminProtocolMethod.java

```diff
@@ -107,7 +111,9 @@ protected <R> Collection<R> invokeConcurrent(Class<R> clazz) throws YarnExceptio
           Pair<SubClusterId, Object> pair = future.get();
           subClusterId = pair.getKey();
           Object result = pair.getValue();
-          results.put(subClusterId, clazz.cast(result));
+          if (result != null) {
+            results.put(subClusterId, clazz.cast(result));
```

Comment:
As we are at it, extract?

Reply:
Thanks for helping to review the code, I will fix it.